crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -245,17 +280,26 @@ async fn mapper_fails_during_sw_update_recovers_and_process_response() -> Result
 
     // Start SM Mapper
     let sm_mapper = start_sm_mapper(broker.port).await?;
-    mqtt_tests::assert_received(
-        &mut responses,
-        TEST_TIMEOUT_MS,
-        vec!["118,software-management\n", "500\n"],
-    )
-    .await;
 
+    let mut msg = responses
+        .recv()
+        .with_timeout(TEST_TIMEOUT_MS)
+        .await
+        .expect_or("No message received before timeout");
+    dbg!(&msg);
+
+    while !msg.contains("500") {
+        msg = responses
+            .recv()
+            .with_timeout(TEST_TIMEOUT_MS)
+            .await
+            .expect_or("No message received before timeout");
+        dbg!(&msg);
+    }
```

Comment:
If the actual logic that you wanted is to keep reading and ignoring messages until that `500` message is seen, wouldn't this be a simpler representation of that logic (using `loop` instead of `while` to avoid that double read and assignment to `msg` before and after starting the loop):
```suggestion
    let mut msg: String;
    loop {
        msg = messages
            .recv()
            .with_timeout(TEST_TIMEOUT_MS)
            .await
            .expect_or("No message received before timeout");
        dbg!(&msg);

        if msg.starts_with("500") {
            break;
        }
    }
    // Here you have the 500 SmartREST message in msg variable which you can assert on
```

Reply:
This will change as soon as `mqtt_test` is updated and should be handled with new function.